core/vault-worker.go

```diff
@@ -11,78 +11,233 @@
  * the License.
  *
- * @author: Tingyu Zeng, Dell
- * @version: 0.1.0
+ * @author: Tingyu Zeng, Dell / Alain Pulluelo, ForgeRock AS
+ * @version: 0.2.2
  *******************************************************************************/
 package main
 
 import (
+	"bytes"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"net/url"
 
 	"github.com/dghubble/sling"
-	"github.com/hashicorp/vault/api"
 )
 
-func initVault(c *api.Sys, path string, inited bool) (string, error) {
+// InitRequest contains a Vault init request regarding the Shamir Secret Sharing (SSS) parameters
+type InitRequest struct {
+	SecretShares    int `json:"secret_shares"`
+	SecretThreshold int `json:"secret_threshold"`
+}
 
-	if inited == false {
-		ir := &api.InitRequest{
-			SecretShares:    1,
-			SecretThreshold: 1,
-		}
+// InitResponse contains a Vault init response
+type InitResponse struct {
+	Keys       []string `json:"keys"`
+	KeysBase64 []string `json:"keys_base64"`
+	RootToken  string   `json:"root_token"`
+}
+
+// UnsealRequest contains a Vault unseal request
+type UnsealRequest struct {
+	Key   string `json:"key"`
+	Reset bool   `json:"reset"`
+}
+
+// UnsealResponse contains a Vault unseal response
+type UnsealResponse struct {
+	Sealed   bool `json:"sealed"`
+	T        int  `json:"t"`
+	N        int  `json:"n"`
+	Progress int  `json:"progress"`
+}
 
-		resp, err := c.Init(ir)
-		r, _ := json.Marshal(resp)
-		ioutil.WriteFile(path, r, 0644)
-		lc.Info(string(r))
-		lc.Info("Vault has been initialized successfully.")
+// ----------------------------------------------------------
+func vaultHealthCheck(config *tomlConfig, httpClient *http.Client) (sCode int, err error) {
 
-		return resp.KeysB64[0], err
-	}
-	s, err := getSecret(path)
+	// Build Vault API full URL
+	url, err := url.Parse(config.SecretService.Scheme + "://" + config.SecretService.Server + ":" + config.SecretService.Port + vaultHealthAPI)
+	// Build Vault HTTP/GET Request
+	jsonBlock := []byte(`{}`)
+	req, err := http.NewRequest(http.MethodGet, url.String(), bytes.NewBuffer(jsonBlock))
+
+	// Prepare the Header
+	req.Header.Set("Content-Type", contentType)
+
+	// GET the request
+	resp, err := httpClient.Do(req)
 	if err != nil {
-		return "", err
+		lc.Error(fmt.Sprintf("Failure sending the Vault health check request: %s", err.Error()))
+		return 0, err
 	}
-	lc.Info("Vault has been initialized previously. Loading the access token for unsealling.")
-	return s.Token, nil
+	defer resp.Body.Close()
+
+	lc.Info(fmt.Sprintf("Vault Health Check HTTP Status: %s (StatusCode: %d)", resp.Status, resp.StatusCode))
+
+	return resp.StatusCode, nil
 }
 
-func unsealVault(c *api.Sys, token string) (bool, error) {
-	if len(token) == 0 {
-		return true, errors.New("error:empty token")
+// ----------------------------------------------------------
+func vaultInit(config *tomlConfig, httpClient *http.Client, debug bool) (sCode int, err error) {
```

Comment:
again I'm not going to validate this function for logical consistency with the vault API's as I'm not familiar with them, but it would be nice to have someone else do this work